src/components/qas/questions/qaCalculators/distanceCalculator/index.jsx

```diff
@@ -0,0 +1,279 @@
+      return this.roundTo(pars, 10);
+    }
+    return '?';
+  };
+
+  solveForMegaParsecs = parsecs => {
+    if (parsecs !== '?') {
+      const megaPars = parsecs * 10 ** 6;
+      return this.roundTo(megaPars, 10);
+    }
+    return '?';
+  };
+
+  solveForLightYears = parsecs => {
+    if (parsecs !== '?') {
+      const ly = parsecs * 3.26;
+      return this.roundTo(ly, 10);
+    }
+    return '?';
+  };
+
+  solveForMegaLightYears = lightYears => {
+    if (lightYears !== '?') {
+      const mly = lightYears / 999315.5373;
+      return this.roundTo(mly, 10);
+    }
+    return '?';
+  };
+
+  roundTo = (number, decimalPlace) => {
+    const roundedNumber = Math.round(number * decimalPlace) / decimalPlace;
+    return roundedNumber;
+  };
+
+  addTheCommas = num =>
+    num.toString().search(/e/gi) < 0
+      ? num.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',')
+      : num;
+
+  render() {
+    const { question, answer, activeId } = this.props;
+    const {
+      hasFocus,
+      answerable,
+      peakMagnitude,
+      distanceModulus,
+      parsecs,
+      megaParsecs,
+      lightYears,
+      megaLightYears,
+    } = this.state;
+    const { id, questionType, label, placeholder } = question;
+    const isTextArea = questionType === 'textArea';
+    const active = activeId === id;
+    const answered = !isEmpty(answer);
+    const cardClasses = classnames('qa-card', { active: hasFocus });
+    const fieldClasses = classnames('qa-text-input', {
+      answered,
+      unanswered: !answered,
+      answerable: answerable || answered || active,
+    });
+
+    return (
+      <Card className={cardClasses}>
+        <CardText>
+          <TextField
+            ref={this.field}
+            id={`text-${isTextArea ? 'area' : 'input'}-${id}`}
+            className={fieldClasses}
+            type="number"
+            min="0"
+            label={<div dangerouslySetInnerHTML={renderDef(label)} />}
+            lineDirection="center"
+            placeholder={placeholder}
+            defaultValue={answered ? answer.content : null}
+            onBlur={this.handleBlur}
+            onFocus={this.handleFocus}
+            onChange={this.handleChange}
+            disabled={!(answerable || answered || active)}
+          />
+        </CardText>
+        <CardActions expander>
+          <p>View Equation and Answers</p>
+        </CardActions>
+        <CardText expandable>
+          <div>
+            <p>DM = {this.addTheCommas(peakMagnitude)} + 19.4</p>
+          </div>
+          <br />
+          <div>
+            <p>
+              d = 10{' '}
+              <span className="exponential-fraction">
+                <span className="numerator">
+                  ({this.addTheCommas(distanceModulus)} + 5)
+                </span>
+                <span className="denominator">/ 5</span>
+              </span>
+            </p>
+          </div>
+          <br />
+          <div>
+            <div>
```

Comment:
Regardless of what you do about my feedback on line 219 - 227, I think these `calculated-measurement`s can be DRYed up.  A method that returns the mark-up you want would be sufficient (passing in the value it should render, the supporting text and units, etc.), but also probably a good opportunity to create a CalculatedMeasurement component.  And that Component is probably not gonna need much more than props and a render method so could be a good chance to get some practice with the functional component syntax.  I recently messed around with a bunch of stuff like this (lil components just for presentational units and values logic) in `components/charts/shared/unit`.  I haven't implemented any of it yet, but we'll have a chance to make sure it works once there's more stuff going on with the graphs.  Take a look if you want functional component inspiration.

Reply:
I like this. Breaking this out into another ticket for Nov12 sprint.